crates/oxc_transformer/src/typescript/annotations.rs

```diff
@@ -222,11 +222,23 @@ impl<'a> TypeScriptAnnotations<'a> {
     }
 
     pub fn transform_simple_assignment_target(&mut self, target: &mut SimpleAssignmentTarget<'a>) {
-        if let Some(expr) = target.get_expression() {
-            if let Expression::Identifier(ident) = expr.get_inner_expression() {
-                // SAFETY: `ast.copy` is unsound! We need to fix.
-                let ident = unsafe { self.ctx.ast.copy(ident) };
-                *target = SimpleAssignmentTarget::AssignmentTargetIdentifier(ident);
+        if let Some(expr) = target.get_expression_mut() {
+            match expr.get_inner_expression_mut() {
+                // `foo!++` to `foo++`
+                Expression::Identifier(ident) => {
+                    *target = self.ctx.ast.simple_assignment_target_from_identifier_reference(
+                        self.ctx.ast.move_identifier_reference(ident),
+                    );
+                }
+                // `foo.bar!++` to `foo.bar++`
+                inner_expr @ match_member_expression!(Expression) => {
+                    *target = SimpleAssignmentTarget::from(
+                        self.ctx.ast.move_member_expression(inner_expr.to_member_expression_mut()),
+                    );
+                }
+                _ => unreachable!(
```

Comment:
It's probably better to push an error and bail, the AST may be dirty here :-/

Reply:
Sounds reasonable